src/python/txtai/pipeline/text/translation.py

```diff
@@ -74,22 +76,28 @@ def __call__(self, texts, target="en", source=None):
         languages = self.detect(values) if not source else [source] * len(values)
         unique = set(languages)
 
-        # Build list of (index, language, text)
-        values = [(x, lang, values[x]) for x, lang in enumerate(languages)]
+        # Build a dict from language to list of (index, text)
+        langdict = defaultdict(list)
```

Comment:
Do we need defaultdict? Can this just be a standard dict `{}`?

Reply:
Actually the ```defaultdict(list)``` marks the dictionary as having values of type list as per what i understand. So that we can directly just call append on the new insertions.

Reply:
If you change line 80 to this and remove the import from line 5, I think this will be good to go. 

```python
        langdict = {}
```

I don't believe there is a usecase for defaultdict.

Reply:
If we change to what you mentioned above, we will get keyerror
```
======================================================================
ERROR: testTranslationWithShowmodels (testtranslation.TestTranslation)
Tests a translation using Marian models and showmodels flag to return
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/yashdatta/Documents/Workspace/txtai/test/python/testpipeline/testtranslation.py", line 97, in testTranslationWithShowmodels
    result = translate(text, "es", showmodels=True)
  File "/Users/yashdatta/Documents/Workspace/txtai/src/python/txtai/pipeline/text/translation.py", line 82, in __call__
    langdict[lang].append((x, values[x]))
KeyError: 'en'
```

This is because, defaultdict is specifically used in situations where we want to generate default values for keys that do not exist. We are using this fact in the loop below

```
        for x, lang in enumerate(languages):
            langdict[lang].append((x, values[x]))
```

because for any lang not existing in the dictionary, an empty list is generated and the value is appended to it.

Does this make sense ?

Reply:
Typically you'd do this first

```python
if lang not in langdict:
  langdict[lang] = []

langdict[lang].append((x, values[x]))
```

That's typically the style as I tend not to use the collections classes unless necessary throughout the code.

You've done a great job, so I'm fine doing that on my end after the merge, it's a minor preference thing.

Reply:
Ah ok understood, I am fine either way, was just wondering what is the reason and if my own understanding is correct.  Internally defaultdict does the exact same thing I think. 
Done, should be good now.

Reply:
I appreciate your diligence in getting this PR done. This is just a stylistic preference to be consistent with the rest of the code.

Thanks again for the PR, if you're interested more, I can point you to others in the queue that would be good to look at.